src/utils.c

```diff
@@ -2497,3 +2497,151 @@ gchar *utils_get_os_info_string(void)
 
 	return os_info;
 }
+
+
+/**
+ * @brief Replaces placeholders in a string with a given filename / path / ...
+ *
+ * This function searches for specified placeholders within the given string
+ * and replaces it with the provided doc information (filename / path / ...)
+ *
+ * @param[in] doc The document from which the name/path replaces the placeholders.
+ *
+ * @param[in] src The original string including the placeholder tokens (...%d...%e...)
+ *
+ * @param[in] needles The string with placeholders to replace.
+ *                    The string is a combination of 'd', 'e', 'f', 'l' and 'p'.
+ *                    'd' : replace %d with the absolute path
+ *                    'e' : replace %e with the filename (excluding extension)
+ *                    'f' : replace %f with the filename (including extension)
+ *                    'l' : replace %l with the current 1-based linenumber
+ *                    'p' : within a project context:
+ *                             replace %p with the absolute project base directory
+ *                          within a document context:
+ *                             replace %p with the absolute path/filename (including extension)
+ *
+ *                    e.g. 'df' will replace all %d and %f placeholders
```

Comment:
What is the goal of this parameter? AFAIK, it is only used to exclude `%l` from the external print command, which doesn't seem very much required to me (I don't see how it would be a problem -- and it could even be seen as useful, e.g. to only print around the caret or some other hypothetical use case).

Reply:
@Pynckels